src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -167,83 +166,86 @@ public override void Dispose()
 
         private async Task ConnectAsyncCore(Uri uri, List<string>? requestedSubProtocols, CancellationToken cancellationToken)
         {
-            object[]? subProtocols = requestedSubProtocols?.ToArray();
-            var onClose = (int code, string reason) =>
+            try
             {
-                if (code != 4000)
+                object[]? subProtocols = requestedSubProtocols?.ToArray();
+                var onClose = (int code, string reason) =>
                 {
-                    _closeStatus = (WebSocketCloseStatus)code;
-                    _closeStatusDescription = reason;
-                    if (_state == WebSocketState.Connecting || _state == WebSocketState.Open)
+                    if (code != 4000)
```

Comment:
maybe make this a named constant or enum (4000 is in the application-defined range)